.github/scripts/update_generation_config.sh

```diff
@@ -1,12 +1,36 @@
 #!/bin/bash
 set -e
 # This script should be run at the root of the repository.
-# This script is used to update googleapis commit to latest in generation
+# This script is used to update parameters in generation
 # configuration at the time of running and create a pull request.
 
 # The following commands need to be installed before running the script:
 # 1. git
 # 2. gh
+# 3. jq
+
+# Utility functions
+function get_major_version() {
+    local key_word=$1
+    major=$(grep "${key_word}" "${generation_config}" | cut -d ':' -f 2 | xargs | cut -d '.' -f 1)
+    echo "${major}"
+}
+
+function get_latest_release_version() {
+    local group_id=$1
+    local artifact_id=$2
+    local major_version=$3
+    latest=$(curl -s "https://search.maven.org/solrsearch/select?q=g:${group_id}+AND+a:${artifact_id}&core=gav&rows=500&wt=json" | jq -r '.response.docs[] | select(.v | test("^[0-9]+(\\.[0-9]+)*$")) | .v' | grep ^"${major_version}". | sort -V | tail -n 1)
```

Comment:
Why do we need a `major_version` to get the latest released version?

Reply:
We don't want to introduce major version bump to the generator in a branch.

Reply:
I'm not sure that's something should be guarded in google-cloud-java. Major version bump of the generator should be rare and it should be guarded in the sdk-platform-java. Assuming there is indeed a major version bump, the automation would not work and we have to manually update the generator version.

Reply:
Same thing for libraries bom version, it is only used for generating README, and I don't think there is a need of guarding libraries bom.

Reply:
This is used in supporting multi-version of gapic generator.

For example, we have two version of gapic generator, v2 and v3. In google-cloud-java, one branch (`branch-a`) is using v2 and another one is using v3. I think we don't want to update the generator to v3 in `branch-a` in this case.

Reply:
I see your point now. I think it is a good consideration and could work well for multi-JDK version scenario, but we probably don't want to guard it this way at this moment because there are other scenarios of multi-version support as well 
1. We could have multiple branches using the same major version but different minor version of gapic generator.
2. We could have multiple branches using the exact same version of gapic generator but different protoc version.
3. We may not want any automated version updates in a branch(LTS).

I think for now we can always update to the latest, and adjust our scripts as needed in the future.

Reply:
OK, I'll remove the major version limitation here. We can adjust the script when new use case arises.

Reply:
Done.